src/main/java/org/spongepowered/common/mixin/api/mcp/block/BlockRailBase_EnumRailDirectionMixin_API.java

```diff
@@ -48,6 +53,78 @@
         return shadow$getName();
     }
 
+    @Override
+    public Optional<Direction> getAscendingDirection() {
+        switch (this.getMetadata()) {
+            case 2:
+                return Optional.of(Direction.EAST);
+            case 3:
+                return Optional.of(Direction.WEST);
+            case 4:
+                return Optional.of(Direction.NORTH);
+            case 5:
+                return Optional.of(Direction.SOUTH);
+            default:
+                return Optional.empty();
+        }
+    }
+
+    @Override
+    public Direction getFirstDirection() {
+        switch (this.getMetadata()) {
+            case 0:
+            case 5:
+            case 8:
+            case 9:
+                return Direction.NORTH;
+            case 1:
+            case 3:
+                return Direction.EAST;
+            case 4:
+            case 6:
+            case 7:
+                return Direction.SOUTH;
+            case 2:
```

Comment:
All of these magic numbers need to be Constants in the `org.spongepowered.common.util.Constants`, whether it's in a new nested class for Rail Directions or whatever.